hudi-common/src/main/java/org/apache/hudi/common/table/view/AbstractTableFileSystemView.java

```diff
@@ -738,7 +799,9 @@ private String formatPartitionKey(String partitionStr) {
    * @param commitsToReturn Commits
    */
   Stream<FileSlice> fetchLatestFileSliceInRange(List<String> commitsToReturn) {
-    return fetchAllStoredFileGroups().map(fileGroup -> fileGroup.getLatestFileSliceInRange(commitsToReturn))
+    return fetchAllStoredFileGroups()
+        .filter(fileGroup -> !isFileGroupReplacedBeforeAny(fileGroup, commitsToReturn))
```

Comment:
@satishkotha  : As discussed, All the replace filtering needs to move to getXXX() apis as the fetch APIs are only responsible for fetching file slices/base-files from different types of storage.

Reply:
Changed. Please take a look.